### PR TITLE
fixed: make WITH_NDEBUG cmake option work again

### PR DESCRIPTION
**Pull request by akva2**
at some point cmake started adding -DNDEBUG to default release CXX flags.

thus we have to invert logic and use -UNDEBUG to remove instead.

note: this will uncover current issues in opm-simulators.

https://github.com/OPM/opm-simulators/pull/3462 fixed test_glift1
https://github.com/OPM/opm-models/pull/667 fixed SPE5CASE1_DYN

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this please

**Comment by akva2** [comment]
jenkins build this opm-simulators=3461 please

**Comment by akva2** [comment]
@totto82 @bska please help with the remaining failures

**Comment by akva2** [comment]
the spe5 one seems to be caused by evaluation of the oil viscosity table outside bounds with extrapolation disabled. why this would happen is well beyond my understanding of things.

**Comment by totto82** [comment]
> the spe5 one seems to be caused by evaluation of the oil viscosity table outside bounds with extrapolation disabled. why this would happen is well beyond my understanding of things.

We have seen this before. My guess is that we for some unconverging iterations gets very nonphysical values for instance for pressure. Did adding the extrapolation flag fix the issue?

**Comment by akva2** [comment]
uhm, i did not try anything apart from running debugger to try to see what's up.

**Comment by akva2** [comment]
jenkins build this opm-models=667 please

**Comment by akva2** [comment]
jenkins build this opm-simulators=3463 please

File: cmake/Modules/UseOptimization.cmake
Comment (blattms) [inline cmake/Modules/UseOptimization.cmake:68]: Shouldn't we rather just remove -DNDEBUG from flags using `list(REMOVE_ITEM flags -DNDEBUG})` in a macro remove_options or similar?

Comment (akva2) [reply cmake/Modules/UseOptimization.cmake:68]: maybe. i did not want to deal with the snake pit that is flags variables (they are usually, but not always, a string rather than a proper list).

**Review by blattms — APPROVED** [review]
Approved if jenkins agrees. Was not aware there was `-U` until this PR.

**Comment by akva2** [comment]
cannot merge it yet, there are still an issue in opm-simulators.

**Comment by akva2** [comment]
jenkins build this please

**Comment by akva2** [comment]
green and approved.

**Pull request merged**
